src/sonic-yang-models/yang-models/sonic-nvgre-tunnel.yang

```diff
@@ -0,0 +1,106 @@
+module sonic-nvgre-tunnel {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-nvgre-tunnel";
+    prefix nvgre;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+    }
+
+    description "NVGRE Tunnel YANG Module for SONiC OS";
+
+    revision 2021-10-31 {
+        description
+            "First Revision";
+    }
+
+    container sonic-nvgre-tunnel {
+
+        container NVGRE_TUNNEL {
+
+            description "NVGRE_TUNNEL part of config_db.json";
+
+            list NVGRE_TUNNEL_LIST {
+
+                key "tunnel_name";
+
+                leaf tunnel_name {
+                    description "NVGRE Tunnel name";
+
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf src_ip {
+                    description "Source IP address";
+
+                    mandatory true;
+                    type inet:ip-address;
+                }
+
+            }
+            /* end of NVGRE_TUNNEL_LIST */
+
+        }
+        /* end of container NVGRE_TUNNEL */
+
+        container NVGRE_TUNNEL_MAP {
+
+            description "NVGRE_TUNNEL_MAP part of config_db.json";
+
+            list NVGRE_TUNNEL_MAP_LIST {
+
+                key "tunnel_name vlan_name";
+
+                leaf tunnel_name {
+                    description "NVGRE Tunnel name";
+
+                    type leafref {
+                        path /nvgre:sonic-nvgre-tunnel/nvgre:NVGRE_TUNNEL/nvgre:NVGRE_TUNNEL_LIST/nvgre:tunnel_name;
+                    }
+                }
+
+                leaf vlan_name {
+                    description "VLAN name";
+
+                    type leafref {
+                        path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name;
+                    }
+                }
+
+                leaf vlan_id {
```

Comment:
Please have either vlan_name or vlan_id, keeping the duplicating information in the YANG doesnt make any sense.

Reply:
This `Config DB` node design - `vlan_name` and `vlan_id` was added to be consistent with the existing `VxLAN Tunnel` feature.
Also, the user can set to `vlan_name` anything he wants, because it is part of the key (`NVGRE_TUNNEL_MAP`), but in the `orchagent` implementation I'm using `vlan_id` because it is consistent with the `SAI` model.

Reply:
Ok, let's please replace vlan_name with just name for better clarity.

Reply:
Fixed